src/server/components/proxyMiddleware/router.ts

```diff
@@ -0,0 +1,81 @@
+import type { NextFunction, Request, Response } from "express";
+import { addCsrfTokenToLocals, singleRouteCsrf } from "server/middlewares/csrf";
+import { countriesList } from "server/services/metadata";
+import { validateCountryLower } from "server/models/listItem/providers/helpers";
+import express from "express";
+import { listExists } from "server/components/proxyMiddleware/helpers";
+import { json, urlencoded } from "body-parser";
+import cookieParser from "cookie-parser";
+import { checkCountryQuestionAnswered } from "./middlewares/checkCountryQuestionAnswered";
+
+/**
+ * proxy middleware does not work if bodyParser, cookies and csrf have been applied to the server before the proxies
+ * are initialised. By applying these middlewares to individual routes, it does not interfere with the proxy.
+ */
+const bodyParser = [json(), urlencoded({ extended: true })];
+const cookies = cookieParser();
+const middleware = [...bodyParser, cookies, singleRouteCsrf, addCsrfTokenToLocals];
+
+export const applyRouter = express.Router();
+
+declare module "express-session" {
+  export interface SessionData {
+    application: {
+      type?: "lawyers" | "funeral-directors" | "translators-interpreters";
+      country?: string;
+      isInitialisedSession?: boolean;
+    };
+  }
+}
```

Comment:
What's the benefit of this code?

Reply:
Probably makes more sense to declare the session type closer to where it's being used.. atm I think it's near the auth stuff